pkg/api/types.go

```diff
@@ -2070,10 +2070,41 @@ const (
 	StrategicMergePatchType PatchType = "application/strategic-merge-patch+json"
 )
 
+// Type and constants for component type.
+type ComponentType string
+
+// These are the valid conditions for the component type.
+const (
+	ComponentAPIServer         ComponentType = "apiserver"
+	ComponentControllerManager ComponentType = "controller-manager"
```

Comment:
However, if we break up controller-manager into multiple binaries, I would recommend making a type for each binary. The type may be useful in the future for knowing what endpoints the component has. For now, the only endpoint components are expected to have is `/healthz`.